File: Web/ClientApp/src/app/domain/entities/manufacturer.ts
import { Employee } from ".";

export class Manufacturer
{
    public id: number;
    public name: string;
    public country: string;
    public address : string;
    public description: string;
    
    public employeeId: number;
    public employee: Employee;

    public static fromJS(data: any = {}): Manufacturer {
        const entity = new Manufacturer();
        
        entity.id = data.id
        entity.name = data.name
        entity.country = data.country
        entity.address = data.address
        entity.description = data.description
        entity.employeeId = data.employeeId
        entity.employee = data.employee

        return entity;
    }
}

File: Web/Models/Manufacturer.cs
namespace Web.Models
{
    public class Manufacturer
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Country { get; set; }
        public string Address { get; set; } 
        public string Description { get; set; }
        
        public int EmployeeId { get; set; }
        public Employee Employee { get; set; }
    }
}